src/components/Connection/Desktop/ImportAccount.vue

```diff
@@ -0,0 +1,291 @@
+      <s-button
+        @click="nextStep"
+        key="step1"
+        :disabled="disabledNextStep"
+        class="s-typography-button--large login-btn"
+        type="primary"
+      >
+        {{ t('desktop.button.next') }}
+      </s-button>
+    </template>
+    <template v-else-if="step === LoginStep.ImportCredentials">
+      <s-form class="login__inputs">
+        <s-input
+          :placeholder="t('desktop.accountName.placeholder')"
+          v-model="accountName"
+          :disabled="loading"
+          :readonly="readonlyAccountName"
+        ></s-input>
+        <s-input
+          :type="inputType"
+          :placeholder="t('desktop.password.placeholder')"
+          v-model="accountPassword"
+          :disabled="loading"
+        >
+          <s-icon :name="iconPasswordStyle" class="eye-icon" size="18" slot="suffix" @click.native="toggleVisibility" />
+        </s-input>
+      </s-form>
+
+      <s-button
+        @click="importAccount"
+        key="step2"
+        :disabled="disabledImportStep"
+        class="s-typography-button--large login-btn"
+        type="primary"
+      >
+        {{ t('desktop.button.importAccount') }}
+      </s-button>
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+import { Mixins, Component, Prop, Ref } from 'vue-property-decorator';
+import { mnemonicValidate } from '@polkadot/util-crypto';
+import { api } from '@sora-substrate/util';
+import LoadingMixin from '@/components/mixins/LoadingMixin';
+import TranslationMixin from '../../mixins/TranslationMixin';
+import { PolkadotJsAccount, KeyringPair$Json } from '../../../types/common';
+import { parseJson } from '../../../util';
+import { LoginStep } from '../../../consts';
+import { getter, action } from '../../../store/decorators';
+
+@Component
+export default class ImportAccount extends Mixins(TranslationMixin, LoadingMixin) {
+  @Prop({ type: String }) readonly step!: LoginStep;
+
+  @getter.account.polkadotJsAccounts polkadotJsAccounts!: Array<PolkadotJsAccount>;
+
+  @action.account.importPolkadotJs importPolkadotJs!: (address: string) => Promise<void>;
+  @action.account.getPolkadotJsAccounts getPolkadotJsAccounts!: () => Promise<void>;
+
+  @Ref('fileInput') readonly fileInput!: HTMLInputElement;
+
+  LoginStep = LoginStep;
+
+  readonly PHRASE_LENGTH = 12;
+
+  mnemonicPhrase = '';
+  accountName = '';
+  accountPassword = '';
+
+  json: Nullable<KeyringPair$Json> = null;
+
+  hiddenInput = true;
+  readonlyAccountName = false;
+
+  get title(): string {
+    switch (this.step) {
+      case LoginStep.Import:
+        return this.t('desktop.heading.importTitle');
+      case LoginStep.ImportCredentials:
+        return this.t('desktop.heading.accountDetailsTitle');
+      default:
+        return '';
+    }
+  }
+
+  get disabledNextStep(): boolean {
+    return this.mnemonicPhrase.length === 0;
+  }
+
+  get disabledImportStep(): boolean {
+    return !(!!this.accountName && !!this.accountPassword);
```

Comment:
5 operators are a lot. empty string && empty string = empty string. !empty string = true

`return !(this.accountName && this.accountPassword);`